### ISSUE TITLE
Fatal Exception: java.util.ConcurrentModificationException

### ISSUE THREAD

**Issue by DimaKoz** [OneSignal/OneSignal-Android-SDK#42]
I got crash in OneSignal-Android-SDK 2.3.0. 
You can see it in [log.txt](https://github.com/one-signal/OneSignal-Android-SDK/files/239212/log.txt).
About device:
<img width="943" alt="screen shot 2016-04-27 at 22 29 57" src="https://cloud.githubusercontent.com/assets/3092276/14865532/1ccd07f4-0cd0-11e6-8ef5-b00f6f9785f4.png">

**Comment by jkasten2** [comment]
@DimaKoz Thanks for the crash report. It looks like there might be an issue with the SDK not supporting access from multiple threads.

Can you confirm if your using OneSignal on multiple threads in your app? Also please provide any code you can where you are calling OneSignal in your app.

**Comment by DimaKoz** [comment]
@jkasten2 
You wrote:
> Can you confirm if your using OneSignal on multiple threads in your app?

I think no.

Below is all the code fragments, where I used by your SDK.

- AndroidManifest.xml:
    
```xml
<application
        android:name=....>
<!--...-->
<meta-data android:name="com.onesignal.NotificationOpened.DEFAULT" android:value="DISABLE" />
<!--...-->
</application>
```

- ActivityMain.java:

```java
    
    //...
    @Override
	protected void onResume() {
		super.onResume();
                //...
		OneSignal.setSubscription(isEnableNotification);
                //...
	}
```
- CustomApplication.java:
```java
      @Override
	public void onCreate() {
            super.onCreate();
            //...
            if (BuildConfig.DEBUG) {
                OneSignal.setLogLevel(OneSignal.LOG_LEVEL.DEBUG, OneSignal.LOG_LEVEL.WARN);
            }
            OneSignal.startInit(this)
                    .setNotificationOpenedHandler(new NotificationHandler())
                    .setAutoPromptLocation(true)
                    .init();
           OneSignal.sendTag(tagKey1, tagValue1);
           // We send six tags
           OneSignal.sendTag(tagKey6, tagValue6);
          //...
      }
```

 - NotificationHandler.java:
```java
  @Override
    public void notificationOpened(String message, JSONObject additionalData, boolean isActive) {

            if (additionalData != null) {
               // we send an `Intent` here
            }
     }
```

**Comment by jkasten2** [comment]
@DimaKoz Thanks for the details. It might be an issue with the number of calls to setSubscription since it is in the `onResume` method of your app. I'll investigate if a number of calls to this in suggestion can trigger this crash.

Are you able to reproduce this error your self?

**Comment by DimaKoz** [comment]
No. 
And `setSubscription` was placed into `try{} catch{}` block

**Comment by jkasten2** [comment]
Fixed in version 2.4.1
https://github.com/one-signal/OneSignal-Android-SDK/releases/tag/2.4.1

Same as issue #58
